### ISSUE TITLE
JS variable renaming breaks scripts

### ISSUE THREAD

**Issue by Simonmicro** [tdewolff/minify#326]
minify triggers an `Unexpected identifier` syntax error with the code below...

```javascript
var a, b, c, d, e, f, g, h, i, j, k, l, m, n, o, p, q, r, s, t, u, v, w, x, y, z;
var something = 42;
something++;
for (let i = 0; i < 10; i++) {
    let badboy = 10;
    badboy++;
}
```
It results in...
```javascript
var something=42,b,c,d,e,f,g,h,i,j,k,l,m,n,o,p,q,r,s,t,u,v,w,x,y,z,a;something++;for(let a=0;a<10;a++)let b=10,b++
```
Note that the variable `a` is both used in the `var` AND `let` scope and causes an syntax error ([test it here](https://esprima.org/demo/validate.html))!

Or to put it into the tools own words:
```
ERROR: cannot minify /app/src/examplenew.js: unexpected b in expression on line 1 and column 107
    1: ...omething++;for(let a=0;a<10;a++)let b=10,b++
```
When I don't mix `var` and `let` it works fine...

I must admit that we use this wonderful tool also at work and yeah... It broke our release version for about 10 minutes - very not good...

**Comment by tdewolff** [comment]
Should be fixed in v2.9.4

>  It broke our release version for about 10 minutes - very not good...

You can pay for support if this is critical to your product. Otherwise, I'm happy to accept PRs for bug fixes!

**Comment by Simonmicro** [comment]
> Should be fixed in v2.9.4

Thank you for your _very_ fast response and fix!

> Otherwise, I'm happy to accept PRs for bug fixes!

I guess I'll need to learn Go... :rofl:
